cloudinit/ssh_util.py

```diff
@@ -240,25 +240,47 @@ def render_authorizedkeysfile_paths(value, homedir, username):
         value = "%h/.ssh/authorized_keys"
     paths = value.split()
     rendered = []
+    user_authorized = default_auth
     for path in paths:
-        for macro, field in macros:
-            path = path.replace(macro, field)
+        for i in range(len(macros)):
+            macro = macros[i][0]
+            field = macros[i][1]
+            path_replaced = path.replace(macro, field)
+            # if %h or %u in path, it means it is a local key
+            if i < 2 and macro in path:
+                user_authorized = path_replaced
+            path = path_replaced
         if not path.startswith("/"):
             path = os.path.join(homedir, path)
         rendered.append(path)
-    return rendered
+    return rendered, user_authorized
 
 
 def extract_authorized_keys(username, sshd_cfg_file=DEF_SSHD_CFG):
     (ssh_dir, pw_ent) = users_ssh_info(username)
     default_authorizedkeys_file = os.path.join(ssh_dir, 'authorized_keys')
+    user_authorizedkey = default_authorizedkeys_file
+    pw_len = len(pw_ent.pw_dir)
     auth_key_fns = []
     with util.SeLinuxGuard(ssh_dir, recursive=True):
         try:
             ssh_cfg = parse_ssh_config_map(sshd_cfg_file)
-            auth_key_fns = render_authorizedkeysfile_paths(
-                ssh_cfg.get("authorizedkeysfile", "%h/.ssh/authorized_keys"),
-                pw_ent.pw_dir, username)
+            auth_key_fns, user_authorizedkey = \
+                render_authorizedkeysfile_paths(
```

Comment:
I think that everything after this call can go outside of the with/try block.

Reply:
Agree